### ISSUE TITLE
Grouping items

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#395]
I've read this great blog post by @mojodna the other day: https://aws.amazon.com/de/blogs/big-data/querying-openstreetmap-with-amazon-athena/

Case study 3 highlights a GROUP BY use case, where the number of buildings is shown along with the building:condition. Currently, it's only possible to return all buildings via Overpass, without the grouping. 

```C
[out:csv("building:condition"; false)]
[bbox:23.6948,90.3907,23.7248,90.4235];
way["building:condition"];
out;
```
(runtime: 92ms)

Running the result through `sort | uniq -c ` will return the desired result.

So, counting is of course possible. However, identifying each different building condition and counting the objects for each of those conditions doesn't seem to be feasible as of today.

![image](https://d2908q01vomqb2.cloudfront.net/b6692ea5df920cad691c20319a6fffd7a4a766b8/2017/04/13/osm_10.gif)

Minimal approach:

- New statement to remove duplicate entries based on a number of given keys, or maybe on some evaluator result (generate derived objects may be another approach)
- Extend `foreach`  to handle derived objects as well

```C
[bbox:23.6948,90.3907,23.7248,90.4235];
way["building:condition"]->.buildings;
remove_duplicates key = t["building:condition"];    // or something similar, maybe based on evaluators to produce a key?!
foreach -> .distinct_entry;

   way.buildings(if:t["building:condition"] == .distinct_entry..);
   make count key = .key....
              count = ...
     
   out;
);

```

**Comment by mmd-osm** [comment]
Similar:

Gibt es bereits ein Tool (/eine overpass-Abfrage?) o.ä., alle verwendeten keys/tags für einen bestimmten Bereich (Ort; bbox) sich in einer schlichten Liste nur mit der Zahl ihres Vorkommens anzeigen/'auswerfen' zu lassen?
Etwa nach dem Schema für Berlin:
amenity=drinking_water -> 66x

**Comment by tyrasd** [comment]
looks like this is implemented in 0.7.55: https://github.com/drolbr/Overpass-API/issues/467#issue-306228173 ?

**Comment by mmd-osm** [comment]
> looks like this is implemented in 0.7.55:

Yes, exactly, @mojodna's example 3 can be written as follows now:

```C
[out:csv(count, "building:condition"; true)]
[bbox:23.6948,90.3907,23.7248,90.4235];
way[building];
for(t["building:condition"])
{
   make stat "building:condition"=_.val,
             count = count(ways);
   out;
};
```

Result:

```
count	building:condition
29500	
4898	average
1785	good
2420	poor

```
